drivers/rtc/rtc_ifx_cat1.c

```diff
@@ -0,0 +1,337 @@
+static cy_stc_syspm_callback_t _ifx_cat1_rtc_pm_cb = {
+	.callback = &_ifx_cat1_rtc_syspm_callback,
+	.type = CY_SYSPM_DEEPSLEEP,
+	.callbackParams = &_ifx_cat1_rtc_pm_cb_params,
+};
+#endif /* CONFIG_PM */
+
+#define _IFX_CAT1_RTC_WAIT_ONE_MS() Cy_SysLib_Delay(_IFX_CAT1_RTC_RETRY_DELAY_MS);
+
+/* Internal macro to validate RTC year parameter falls within 21st century */
+#define IFX_CAT1_RTC_VALID_CENTURY(year) ((year) >= _IFX_CAT1_RTC_INIT_CENTURY)
+
+#define MAX_IFX_CAT1_CAL (60)
+
+/* Convert parts per billion to groupings of 128 ticks added or removed from one hour of clock
+ * cycles at 32768 Hz.
+ *
+ * ROUND_DOWN(ppb * 32768Hz * 60min * 60sec / 1000000000, 128) / 128
+ * ROUND_DOWN(ppb * 117964800 / 1000000000, 128) / 128
+ * ROUND_DOWN(ppb * 9216 / 78125, 128) / 128
+ */
+#define PPB_TO_WCO_PULSE_SETS(ppb) ((ROUND_DOWN((ppb * 9216 / 78125), 128)) / 128)
+
+/* Convert groupings of 128 ticks added or removed from one hour of clock cycles at
+ * 32768 Hz to parts per billion
+ *
+ * wps * 128 * 1000000000 / 32768Hz * 60min * 60sec
+ * wps * 128000000000 / 117964800
+ * wps * 78125 / 72
+ */
+#define WCO_PULSE_SETS_TO_PPB(wps) (wps * 78125 / 72)
+
+struct ifx_cat1_rtc_config {
+	uint32_t irqn;
+};
+
+struct ifx_cat1_rtc_data {
+	struct k_spinlock lock;
+};
+
+static inline uint16_t _ifx_cat1_rtc_get_state(void)
+{
+	return _FLD2VAL(_IFX_CAT1_RTC_BREG_STATE, _IFX_CAT1_RTC_BREG);
+}
+
+static inline void _ifx_cat1_rtc_set_state(uint16_t init)
+{
+	_IFX_CAT1_RTC_BREG &= _IFX_CAT1_RTC_BREG_CENTURY_Msk;
+	_IFX_CAT1_RTC_BREG |= _VAL2FLD(_IFX_CAT1_RTC_BREG_STATE, init);
+}
+
+static inline uint16_t _ifx_cat1_rtc_get_century(void)
+{
+	return _FLD2VAL(_IFX_CAT1_RTC_BREG_CENTURY, _IFX_CAT1_RTC_BREG);
+}
+
+static inline void _ifx_cat1_rtc_set_century(uint16_t century)
+{
+	_IFX_CAT1_RTC_BREG &= _IFX_CAT1_RTC_BREG_STATE_Msk;
+	_IFX_CAT1_RTC_BREG |= _VAL2FLD(_IFX_CAT1_RTC_BREG_CENTURY, century);
+}
+
+static void _ifx_cat1_rtc_from_pdl_time(cy_stc_rtc_config_t *pdlTime, const int year,
+					struct rtc_time *z_time)
+{
+	CY_ASSERT(pdlTime != NULL);
+	CY_ASSERT(z_time != NULL);
+
+	z_time->tm_sec = (int)pdlTime->sec;
+	z_time->tm_min = (int)pdlTime->min;
+	z_time->tm_hour = (int)pdlTime->hour;
+	z_time->tm_mday = (int)pdlTime->date;
+	z_time->tm_mon = (int)(pdlTime->month - 1u);
+	z_time->tm_year = (int)(year - _IFX_CAT1_RTC_TM_YEAR_BASE);
+	z_time->tm_wday = (int)(pdlTime->dayOfWeek - 1u);
+
+	/* year day not known in pdl RTC structure without conversion */
+	z_time->tm_yday = -1;
+
+	/* daylight savings currently marked as unknown */
+	z_time->tm_isdst = -1;
+
+	/* nanoseconds not tracked by ifx code. Set to value indicating unknown */
+	z_time->tm_nsec = 0;
+}
+
+static void _ifx_cat1_rtc_isr_handler(void)
+{
+	Cy_RTC_Interrupt(_ifx_cat1_rtc_dst, NULL != _ifx_cat1_rtc_dst);
+}
+
+void _ifx_cat1_rtc_century_interrupt(void)
+{
+	_ifx_cat1_rtc_set_century(_ifx_cat1_rtc_get_century() + 100);
```

Comment:
Why are we adding 100 here? Worthy of comment.

Reply:
I will add a comment to reflect this is the code.